File: leetcode/1.Tow_Sum/java/TwoSum.java
public class TwoSum {

    private static int[] towSum(int[] nums, int target) {
        for(int i = 0; i < nums.length; i++) {
            for(int j = i +1; j < nums.length; ++j) {
                if(nums[i] + nums[j] == target)
                    return new int[] {i, j};
            }
        }
        return new int[0];
    }

    public static void main(String[] args) {
        int[] myList = {1, 3, 5, 7, 9};
        int target = 4;
        int[] result = towSum(myList, target);
        System.out.println(result[0]);
        System.out.println(result[1]);
    }
}

File: leetcode/1.Tow_Sum/python/twoSum.py
#!/usr/bin/env python

def twoSum(nums, target):
    hash_map = dict()
    for i, x in enumerate(nums):
        if target - x in hash_map:
            return [i, hash_map[target -x]]
        hash_map[x] = i

nums = [2,7,11,15]
target = 9
print(twoSum(nums, target))
